Added Tutorial: Gloo Edge for #3199

Parsifal-M opened the pull request:
Signed-off-by: Peter Macdonald <macdonald.peter90@gmail.com>

<!--

Thanks for submitting a PR to OPA!

Before pressing 'Create pull request' please read the checklist below.

* All code changes should be accompanied with tests. If you are not
modifying any tests, just provide a short explanation of why updates
to tests are not necessary. In addition to helping catch bugs, tests
are extremely helpful in providing _context_ that explains how your
changes can be used.

* All changes to public APIs **must** be accompanied with
docs. Examples of public APIs include built-in functions,
config fields, and of course, exported Go types/functions/constants/etc.

* Commit messages should explain _why_ you made the changes, not what
you changed. Use active voice. Keep the subject line under 50
characters or so.

* All commits must be signed off by the author. If you are not
familiar with signing off, see our contributor guide below.

For more information on contributing to OPA see:

* [Contributing Guide](https://www.openpolicyagent.org/docs/latest/contributing/)
  for high-level contributing guidelines and development setup.

-->

Hey @ashutosh-narkar I've added the tutorial as requested, I did some very small changes to some of the wording (i hope you don't mind). Also I noticed it doesn't have a "Wrapping Up" section, is this something you'd like to add? if you have something in  mind I can add this also.

Please let me know your feedback? :smile: 

Thanks!

ashutosh-narkar commented on docs/content/envoy-tutorial-gloo-edge.md:
````diff
@@ -0,0 +1,242 @@
+
+Store the policy in Kubernetes as a Secret.
+
+```bash
````
We should host the policy on a local bundle server. See [this](https://www.openpolicyagent.org/docs/latest/envoy-tutorial-standalone-envoy/#4-publish-opa-bundle) for reference.

ashutosh-narkar commented on docs/content/envoy-tutorial-gloo-edge.md:
```diff
@@ -0,0 +1,242 @@
+            - "--set=plugins.envoy_ext_authz_grpc.query=data.envoy.authz.allow"
+            - "--set=decision_logs.console=true"
+            - "--ignore=.*"
+            - "/policy/policy.rego"
```
Update this to pull policy from the bundle server. See [this](https://www.openpolicyagent.org/docs/latest/envoy-tutorial-standalone-envoy/#5-create-app-deployment-with-opa-and-envoy-sidecars) for reference.

ashutosh-narkar commented:
@Parsifal-M thanks for working on this. It would be great if we had a `Wrap Up` section like the other tutorials.

Parsifal-M commented:
> @Parsifal-M thanks for working on this. It would be great if we had a `Wrap Up` section like the other tutorials.

Hey @ashutosh-narkar 

Will check on your reviews and add a Wrap Up section :+1: 

Thanks!

Parsifal-M commented:
Hey @ashutosh-narkar just FYI I have not forgotten, I'm on holiday this week but I did bring my laptop to do some cool bits and pieces, so I will post back later tonight I think 👍

Parsifal-M commented:
Hello @ashutosh-narkar

Sorry for the delay on this :smile: I actually noticed I missed a small section while testing this as well which I have corrected.

Tested the whole tutorial locally and I seem to get the correct results.

Please let me know if we need to update anything further? I have squashed the commits also.

Thanks!

Parsifal-M commented:
Need to sort out my branch sorry, will commit again in a bit...
